src/Analyzers/RoslynExtensions.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.DurableTask.Analyzers;
+
+static class RoslynExtensions
+{
+    public static bool TryGetSingleValueFromAttribute<T>(this ISymbol? symbol, INamedTypeSymbol attributeSymbol, out T value)
+    {
+        if (symbol.TryGetConstructorArgumentsFromAttribute(attributeSymbol, out ImmutableArray<TypedConstant> constructorArguments))
+        {
+            object? valueObj = constructorArguments.FirstOrDefault().Value;
+            if (valueObj != null)
+            {
+                value = (T)valueObj;
+                return true;
+            }
+        }
+
+        value = default!;
+        return false;
+    }
+
+    public static bool TryGetConstructorArgumentsFromAttribute(this ISymbol? symbol, INamedTypeSymbol attributeSymbol, out ImmutableArray<TypedConstant> constructorArguments)
+    {
+        if (symbol != null)
+        {
+            foreach (AttributeData attribute in symbol.GetAttributes())
+            {
+                if (attributeSymbol.Equals(attribute.AttributeClass, SymbolEqualityComparer.Default))
+                {
+                    constructorArguments = attribute.ConstructorArguments;
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public static bool ContainsAttributeInAnyMethodArguments(this IMethodSymbol methodSymbol, INamedTypeSymbol attributeSymbol)
+    {
+        return methodSymbol.Parameters
+            .SelectMany(p => p.GetAttributes())
+            .Any(a => attributeSymbol.Equals(a.AttributeClass, SymbolEqualityComparer.Default));
+    }
+
+    public static bool ImplementsInterface(this INamedTypeSymbol symbol, ISymbol interfaceSymbol)
+    {
+        return symbol.AllInterfaces.Any(i => interfaceSymbol.Equals(i, SymbolEqualityComparer.Default));
+    }
+
+    public static bool InheritsFromOpenGeneric(this INamedTypeSymbol symbol, ITypeSymbol type)
```

Comment:
Is the "OpenGeneric` naming accurate? The implementation doesn't look like it necessarily has to be an open generic.

Reply:
You are right, I renamed it to: `BaseTypeIsConstructedFrom`. Then its usage becomes:
```cs
if (!classSymbol.BaseTypeIsConstructedFrom(knownSymbols.TaskOrchestratorBaseClass))
{
    return;
}
```

Thoughts?

Reply:
Sounds good - `InheritsFrom` is also a good name,